README.md

```diff
@@ -44,10 +45,162 @@ thrift_files: Mix.Utils.extract_files(["thrift"], [:thrift])
+``` 
+
+The client supports the following options:
+
+Option name      |  Type | Description
+-----------------|-------|-------------
+`:tcp_opts` | keyword | A keyword list of tcp options
+`:gen_server_opts` | keyword | A keyword list of options for the gen server 
+
+##### TCP Opts 
+
+Name             | Type | Description 
+-----------------|------|---------------
+`:timeout`       | positive integer | The default timeout for reading from, writing to and connecting to sockets.
+`send_timeout`   | positive integer | The amount of time in milliseconds to wait before sending data fails.
+`backoff_calculator` | (int) -> int | A single argument function that takes the number of retries and returns the amount of time to wait in milliseconds before reconnecting. The default implementation waits 100, 100, 200, 300, 500, 800 and then 1000 ms. All retries after that will wait 1000ms.
 
-## Thrift IDL Parsing
 
-This package also contains experimental support for parsing [Thrift IDL][idl]
+##### GenServer Opts 
+Name             | Type | Description 
+-----------------|------|---------------
+`timeout`        | A positive integer | The amount of time in milliseconds the Client's GenServer waits for a reply. After this, the GenServer will exit with `{:error, :timeout}`.
+
+
+### Example of using options
+
+```elixir
+alias Thrift.Test.UserService.Clients.Binary.Framed, as: Client
+{:ok, client} = UserService.Clients.Binary.Framed.start_link("localhost", 2345, 
+                tcp_opts: [backoff_calculator: fn(retry_count) -> retry_count * 1000], gen_server_opts: [timeout: 10_000])
+                
+```
+In the above example, the client will use a (very) conservative backoff calculator that waits an additional second each time the client retries. In other words, the first retry will wait 1 second, the second will wait 2 seconds, and the third will wait 3. It sets the GenServer timeout to be ten seconds, which means the remote side can take its time to reply. 
+
+
+## Using The Server
+Creating a thrift server is slightly more involved than creating the client, because you need to create a module to handle the work. Fortunately, a `Behaviour` for this module, complete with correct success typing, is generated for you. To implement this behaviour, use the `@behaviour` module attribute. The compiler will now inform you about any missed functions.
+
+```elixir
+defmodule UserServiceHandler do 
+  @behaviour Thrift.Test.UserService.Handler.Behaviour
+  
+  def ping, do: true 
+  
+  def get_user_by_id(user_id) do 
+	case Backend.find_user_by_id(user_id) do 
+		{:ok, user} -> 
+			user 
+		{:error, _} -> 
+			raise Thrift.Test.UserNotFound.exception message: "could not find user with id #{user_id}"
+    end
+  end
+  
+  def delete_user(user_id) do 
+    case Backend.delete_user(user_id) do 
+	   :ok -> true 
+	   _   -> false
+  end
+end
+```
+
+
+After you have your handler module, you use it to build your server. 
```

Comment:
This is a weird paragraph... 

Suggestion: "To start a server with `UserServiceHandler` as the callback module:" <code> "RPC calls to the server are delegated to `UserServiceHandler`."